test/vast/Dialect/HighLevel/glob-linkage-a.c

```diff
@@ -0,0 +1,16 @@
+// RUN: %vast-front %s -vast-emit-mlir=hl -o - | %file-check %s
+//
+// CHECK: hl.var @ew, <extern_weak>
+extern int __attribute__((weak)) ew;
+// CHECK: hl.var @wundef, <weak>
+int __attribute__((weak)) wundef;
+// CHECK: hl.var @wdef, <weak>
+int __attribute__((weak)) wdef = 5;
+// CHECK: hl.var @ewdef, <weak>
+extern int __attribute__((weak)) ewdef = 5;
+// CHECK: hl.var @edef, <external>
+extern int edef = 5;
+// CHECK: hl.var @undef, <common>
+int undef;
```

Comment:
From my understanding this is correct (see [clang/llvm](https://llvm.org/docs/LangRef.html#linkage-types) documentation), but when testing this in [godbolt](https://godbolt.org/z/KzKTKzre8) or locally it actually has `external` linkage. Not sure how to proceed with this